src/high-level/allocation.lisp

```diff
@@ -20,14 +20,34 @@
     1. The allocated storage.
     2. A finalizer thunk of type FINALIZER, which should be called when the memory is OK to be freed.
 NOTE: Note that the finalizer may close over the allocated vector."
-  '(function (integer &key (:element-type t) (:initial-element t)) (values (simple-array *) finalizer)))
+  '(function (integer t t) (values (simple-array *) finalizer)))
```

Comment:
This is not really allegro-specific. The motivation is to simplify this low-level interface. `xxx-allocator` will be called by `allocate` function, which is already using keyword arguments.